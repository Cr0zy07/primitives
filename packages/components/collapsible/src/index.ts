export { OkuCollapsible } from './collapsible'
export { OkuCollapsibleTrigger } from './collapsible-trigger'
export { OkuCollapsibleContent } from './collapsible-content'
export { OkuCollapsibleContentImpl } from './collapsible-content-impl'

export type {
  CollapsibleProps,
  CollapsibleEmits,
  CollapsibleElement,
  CollapsibleNativeElement,
} from './props'

export type {
  CollapsibleContentProps,
  CollapsibleContentElement,
  CollapsibleContentNativeElement,
} from './props'

export type {
  CollapsibleContentImplProps,
  CollapsibleContentImplElement,
  CollapsibleContentImplNativeElement,
} from './props'

export type {
  CollapsibleTriggerProps,
  CollapsibleTriggerEmits,
  CollapsibleTriggerElement,
  CollapsibleTriggerNativeElement,
} from './props'

export {
  collapsibleContentProps,
  collapsibleProps,
  collapsibleTriggerProps,
  createCollapsibleScope,
} from './props'
