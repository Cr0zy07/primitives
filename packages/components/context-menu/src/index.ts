import type { } from '@floating-ui/vue'

export { OkuContextMenuArrow } from './context-menu-arrow'
export { OkuContextMenuCheckboxItem } from './context-menu-checkbox-item'
export { OkuContextMenuContent } from './context-menu-content'
export { OkuContextMenuGroup } from './context-menu-group'
export { OkuContextMenuItemIndicator } from './context-menu-item-indicator'
export { OkuContextMenuItem } from './context-menu-item'
export { OkuContextMenuLabel } from './context-menu-label'
export { OkuContextMenuPortal } from './context-menu-portal'
export { OkuContextMenuRadioGroup } from './context-menu-radio-group'
export { OkuContextMenuRadioItem } from './context-menu-radio-item'
export { OkuContextMenuSeparator } from './context-menu-separator'
export { OkuContextMenuSubContent } from './context-menu-sub-content'
export { OkuContextMenuSubTrigger } from './context-menu-sub-trigger'
export { OkuContextMenuSub } from './context-menu-sub'
export { OkuContextMenuTrigger } from './context-menu-trigger'
export { OkuContextMenu } from './context-menu'

export type {
  ContextMenuArrowProps,
  ContextMenuArrowElement,
  ContextMenuArrowNativeElement,
} from './props'

export type {
  ContextMenuCheckboxItemProps,
  ContextMenuCheckboxItemElement,
  ContextMenuCheckboxItemNativeElement,
} from './props'

export type {
  ContextMenuContentProps,
  ContextMenuContentElement,
  ContextMenuContentNativeElement,
} from './props'

export type {
  ContextMenuGroupProps,
  ContextMenuGroupElement,
  ContextMenuGroupNativeElement,
} from './props'

export type {
  ContextMenuItemIndicatorProps,
  ContextMenuItemIndicatorElement,
  ContextMenuItemIndicatorNativeElement,
} from './props'

export type {
  ContextMenuItemProps,
  ContextMenuItemElement,
  ContextMenuItemNativeElement,
} from './props'

export type {
  ContextMenuLabelProps,
  ContextMenuLabelElement,
  ContextMenuLabelNativeElement,
} from './props'

export type {
  ContextMenuPortalProps,
} from './props'

export type {
  ContextMenuRadioGroupProps,
  ContextMenuRadioGroupElement,
  ContextMenuRadioGroupNativeElement,
} from './props'

export type {
  ContextMenuRadioItemProps,
  ContextMenuRadioItemElement,
  ContextMenuRadioItemNativeElement,
} from './props'

export type {
  ContextMenuSeparatorProps,
  ContextMenuSeparatorElement,
  ContextMenuSeparatorNativeElement,
} from './props'

export type {
  ContextMenuSubContentProps,
  ContextMenuSubContentElement,
  ContextMenuSubContentNativeElement,
} from './props'

export type {
  ContextMenuSubTriggerProps,
  ContextMenuSubTriggerElement,
  ContextMenuSubTriggerNativeElement,
} from './props'

export type {
  ContextMenuSubProps,
} from './props'

export type {
  ContextMenuTriggerProps,
  ContextMenuTriggerElement,
  ContextMenuTriggerNativeElement,
} from './props'

export type {
  ContextMenuProps,
} from './props'
