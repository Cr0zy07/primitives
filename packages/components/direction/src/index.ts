export { useDirection, Provider, OkuDirection } from './direction'
