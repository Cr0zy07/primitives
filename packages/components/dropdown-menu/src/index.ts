import type { } from '@floating-ui/vue'

export { OkuDropdownMenuArrow } from './dropdown-menu-arrow'
export { OkuDropdownMenuCheckboxItem } from './dropdown-menu-checkbox-item'
export { OkuDropdownMenuContent } from './dropdown-menu-content'
export { OkuDropdownMenuGroup } from './dropdown-menu-group'
export { OkuDropdownMenuItemIndicator } from './dropdown-menu-item-indicator'
export { OkuDropdownMenuItem } from './dropdown-menu-item'
export { OkuDropdownMenuLabel } from './dropdown-menu-label'
export { OkuDropdownMenuPortal } from './dropdown-menu-portal'
export { OkuDropdownMenuRadioGroup } from './dropdown-menu-radio-group'
export { OkuDropdownMenuRadioItem } from './dropdown-menu-radio-item'
export { OkuDropdownMenuSeparator } from './dropdown-menu-separator'
export { OkuDropdownMenuSubContent } from './dropdown-menu-sub-content'
export { OkuDropdownMenuSubTrigger } from './dropdown-menu-sub-trigger'
export { OkuDropdownMenuSub } from './dropdown-menu-sub'
export { OkuDropdownMenuTrigger } from './dropdown-menu-trigger'
export { OkuDropdownMenu } from './dropdown-menu'

export type {
  DropdownMenuArrowProps,
  DropdownMenuArrowElement,
  DropdownMenuArrowNativeElement,
} from './props'

export type {
  DropdownMenuCheckboxItemProps,
  DropdownMenuCheckboxItemEmits,
  DropdownMenuCheckboxItemElement,
  DropdownMenuCheckboxItemNativeElement,
} from './props'

export type {
  DropdownMenuContentProps,
  DropdownMenuContentEmits,
  DropdownMenuContentElement,
  DropdownMenuContentNativeElement,
} from './props'

export type {
  DropdownMenuGroupProps,
  DropdownMenuGroupElement,
  DropdownMenuGroupNativeElement,
} from './props'

export type {
  DropdownMenuItemIndicatorProps,
  DropdownMenuItemIndicatorElement,
  DropdownMenuItemIndicatorNativeElement,
} from './props'

export type {
  DropdownMenuItemProps,
  DropdownMenuItemEmits,
  DropdownMenuItemElement,
  DropdownMenuItemNativeElement,
} from './props'

export type {
  DropdownMenuLabelProps,
  DropdownMenuLabelElement,
  DropdownMenuLabelNativeElement,
} from './props'

export type {
  DropdownMenuPortalProps,
} from './props'

export type {
  DropdownMenuRadioGroupProps,
  DropdownMenuRadioGroupElement,
  DropdownMenuRadioGroupNativeElement,
} from './props'

export type {
  DropdownMenuRadioItemProps,
  DropdownMenuRadioItemEmits,
  DropdownMenuRadioItemElement,
  DropdownMenuRadioItemNativeElement,
} from './props'

export type {
  DropdownMenuSeparatorProps,
  DropdownMenuSeparatorElement,
  DropdownMenuSeparatorNativeElement,
} from './props'

export type {
  DropdownMenuSubContentProps,
  DropdownMenuSubContentEmits,
  DropdownMenuSubContentElement,
  DropdownMenuSubContentNativeElement,
} from './props'

export type {
  DropdownMenuSubTriggerProps,
  DropdownMenuSubTriggerEmits,
  DropdownMenuSubTriggerElement,
  DropdownMenuSubTriggerNativeElement,
} from './props'

export type {
  DropdownMenuSubProps,
  DropdownMenuSubEmits,
} from './props'

export type {
  DropdownMenuTriggerProps,
  DropdownMenuTriggerEmits,
  DropdownMenuTriggerElement,
  DropdownMenuTriggerNativeElement,
} from './props'

export type {
  DropdownMenuProps,
  DropdownMenuEmits,
} from './props'
