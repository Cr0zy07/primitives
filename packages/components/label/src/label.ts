import { defineComponent, h } from 'vue'
import { useForwardRef } from '@oku-ui/use-composable'
import { Primitive } from '@oku-ui/primitive'
import { LABEL_NAME, labelProps } from './props'
import type { LabelEmits, LabelNativeElement } from './props'

const label = defineComponent({
  name: LABEL_NAME,
  inheritAttrs: false,
  props: {
    ...labelProps.props,
  },
  emits: {
    ...labelProps.emits,
  },
  setup(props, { attrs, slots, emit }) {
    const forwardedRef = useForwardRef()

    return () => h(Primitive.label, {
      ...attrs,
      ref: forwardedRef,
      asChild: props.asChild,
      onMousedown: (event: LabelEmits['mousedown'][0]) => {
        emit('mousedown', event)
        // prevent text selection when double clicking label
        if (!event.defaultPrevented && event.detail > 1)
          event.preventDefault()
      },
    }, () => slots.default?.())
  },
})

// TODO: https://github.com/vuejs/core/pull/7444 after delete
export const OkuLabel = label as typeof label & (new () => { $props: LabelNativeElement })
