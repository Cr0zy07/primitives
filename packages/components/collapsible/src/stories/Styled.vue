<script setup lang="ts">
import { OkuCollapsible, OkuCollapsibleContent, OkuCollapsibleTrigger } from '@oku-ui/collapsible'
</script>

<template>
  <OkuCollapsible class="collapsible">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>
</template>
