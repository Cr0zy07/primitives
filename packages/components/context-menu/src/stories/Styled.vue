<script setup lang="ts">
import { OkuContextMenu, OkuContextMenuContent, OkuContextMenuItem, OkuContextMenuPortal, OkuContextMenuSeparator, OkuContextMenuTrigger } from '@oku-ui/context-menu'
</script>

<template>
  <div
    :style="{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: '200vw',
      height: '200vh',
      gap: '20px',
    }"
  >
    <OkuContextMenu>
      <OkuContextMenuTrigger class="context-menu-trigger">
        Right click here
      </OkuContextMenuTrigger>
      <OkuContextMenuPortal>
        <OkuContextMenuContent class="context-menu-content" :align-offset="-5">
          <OkuContextMenuItem class="context-menu-item" @select="console.log('undo')">
            Undo
          </OkuContextMenuItem>
          <OkuContextMenuItem class="context-menu-item" @select="console.log('redo')">
            Redo
          </OkuContextMenuItem>
          <OkuContextMenuSeparator class="context-menu-separator" />
          <OkuContextMenuItem class="context-menu-item" disabled @select="console.log('cut')">
            Cut
          </OkuContextMenuItem>
          <OkuContextMenuItem class="context-menu-item" @select="console.log('copy')">
            Copy
          </OkuContextMenuItem>
          <OkuContextMenuItem class="context-menu-item" @select="console.log('paste')">
            Paste
          </OkuContextMenuItem>
        </OkuContextMenuContent>
      </OkuContextMenuPortal>
    </OkuContextMenu>
  </div>
</template>
