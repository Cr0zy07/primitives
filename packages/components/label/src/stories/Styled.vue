<script setup lang="ts">
import { OkuLabel } from '@oku-ui/label'
</script>

<template>
  <OkuLabel class="label" @click="console.log">
    Label
  </OkuLabel>
</template>
