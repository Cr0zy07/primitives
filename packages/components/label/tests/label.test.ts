import { enableAutoUnmount, shallowMount } from '@vue/test-utils'
import type { VueWrapper } from '@vue/test-utils'
import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'

import { OkuLabel } from '../src'

import Styled from '../src/stories/Styled.vue'
import WithControl from '../src/stories/WithControl.vue'

enableAutoUnmount(afterEach)

// TODO: add test
describe.skip('okuLabel', () => {
  // eslint-disable-next-line unused-imports/no-unused-vars
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount({
      components: {
        OkuLabel,
      },
      setup() {
        return { }
      },
      template: `
      `,
    }, {
      attachTo: document.body,
    })
  })
})

describe('okuLabel Stories', () => {
  describe('styled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Styled>>

    beforeEach(async () => {
      wrapper = shallowMount(Styled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('withControl', () => {
    let wrapper: VueWrapper<InstanceType<typeof WithControl>>

    beforeEach(async () => {
      wrapper = shallowMount(WithControl, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
