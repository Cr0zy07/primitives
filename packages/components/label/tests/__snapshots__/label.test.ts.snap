// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuLabel Stories > control > should render correctly 1`] = `"<button class=\\"label-control\\"> Control </button>"`;

exports[`okuLabel Stories > styled > should render correctly 1`] = `"<oku-label-stub aschild=\\"false\\" class=\\"label\\"></oku-label-stub>"`;

exports[`okuLabel Stories > withControl > should render correctly 1`] = `
"<h1>Wrapping control</h1>
<oku-label-stub aschild=\\"false\\"></oku-label-stub>
<h1>Referencing control</h1>
<control-stub id=\\"control\\" class=\\"label-control\\"></control-stub>
<oku-label-stub aschild=\\"false\\" for=\\"control\\"></oku-label-stub>"
`;
